app/assets/stylesheets/fae/modules/forms/_base.scss

```diff
@@ -94,6 +94,7 @@ input[type=email] {
 
   // Specificity hack
   &.table-add-link {
+    margin-top: 13px;
```

Comment:
Additional `margin-top` helps reconcile line-height changes as part of the move to `h2` tags from `h3` tags